llvm/lib/Transforms/Vectorize/LoopVectorizationPlanner.h

```diff
@@ -177,8 +177,11 @@ class VPBuilder {
                               Type *ResultTy,
                               std::optional<FastMathFlags> FMFs = {},
                               DebugLoc DL = {}, const Twine &Name = "") {
-    return tryInsertInstruction(new VPInstructionWithType(
-        Opcode, Operands, ResultTy, FMFs.value_or(FastMathFlags()), DL, Name));
+    if (FMFs)
```

Comment:
It's not obvious to me how this is related to `Add VPInstruction::StepVector and use it in VPWidenIntOrFpInductionRecipe`? Is it a requirement for this patch?

Reply:
Yeah this was from the review comment here https://github.com/llvm/llvm-project/pull/129508#discussion_r2052775615, I'll go ahead and split this off

Reply:
I've opened up https://github.com/llvm/llvm-project/pull/137632 for it anyway, happy to close it if preferred

Reply:
VPInstruction::StepVector would be the first VPInstructionWithType that doesn't take FastMathFlags. The current VPBuilder::createNaryOp always calls the VPInstructionWithType with FMFs though, which triggers an assertion if you don't pass in any FMFs i.e. in this PR.

The change then fixes up createNaryOp to choose between the FMF and non-FMF constructor, which fixes the assertion failure